core/hash.rbs

```diff
@@ -935,8 +935,8 @@ class Hash[unchecked out K, unchecked out V] < Object
   #
   # If no block is given, an enumerator is returned instead.
   #
-  def transform_values!: () -> Enumerator[K, Hash[K, untyped]]
-                       | () { (V) -> V } -> Hash[K, V]
+  def transform_values!: () -> Enumerator[V, Hash[K, untyped]]
+                       | [A] () { (V) -> A } -> Hash[K, A]
```

Comment:
Please keep the monomorphic version.

Reply:
Thanks. I'll just wait until the discussion on `collect!` is resolved as it's related.